### PR TITLE
Update API contract

### PR DESCRIPTION
**Pull request by NoelLH**
Also proposes to start migrating `DonorAccount\Create` to an endpoint more consistent with others (supporting both for now).

### CODE REVIEW

File: api.yaml [hunk 1]
```diff
@@ -2,51 +2,94 @@ openapi: 3.0.0
...
+        schema:
+          type: string
+          format: uuid
+        description: Identifier of the donor
+        required: true
+
     post:
-      summary: Creates a donation
-      operationId: addDonation
-      description: Initiates a donation with pending status
```
Comment (bdsl) [inline api.yaml:22]: is there a reason for deleting the mention of the pending status? I don't know if we need to mention the status by name but may be helpful to mention that this is legitimately used when someone is only *considering* donating. "Create a donation" could be read as much more final than that. But maybe it doesn't matter much for the audience of this document.

File: api.yaml [hunk 2]
```diff
@@ -2,51 +2,94 @@ openapi: 3.0.0
...
       requestBody:
         content:
           application/json:
             schema:
               $ref: '#/components/schemas/Donation'
-        description: Donation to create
+
+    get:
+      operationId: listDonations
+      summary: List existing complete donations made by the person
```
Comment (bdsl) [inline api.yaml:60]: do we need to document that it looks like the implementation actually ignores the person ID in the URL and only uses the ID in the JWT header sent with the request? I think this is actually the same for createDonation, and I think I considered not having the person ID in the URL when creating this route for that reason.

**Review by bdsl — APPROVED** [review]

File: api.yaml (hunk 1 above)
Comment (NoelLH) [reply api.yaml:22]: Not a specific one, no. This diff is actually the result of deleting the (non-`personId`) endpoint entirely. I will merge the description in with `createDonation`'s to further clarify.

File: api.yaml (hunk 2 above)
Comment (NoelLH) [reply api.yaml:60]: I think the token verifier & middleware do require a match don't they @bdsl? If I'm reading it correctly then a mismatch gives you
> JWT error: Not authorised for person ID ...

It's true that which person the request is for is over-specified but I think it's worth it to have clearer & more consistent URIs when looking at request logs etc.

File: api.yaml (hunk 1 above)
Comment (NoelLH) [reply api.yaml:22]: Done in 8525d41d53139507a331ed530ba06157bbcfdfc9

**Pull request merged**